src/Symfony/Bundle/FrameworkBundle/Command/WorkflowDumpCommand.php

```diff
@@ -40,19 +42,28 @@ class WorkflowDumpCommand extends Command
      *
      * @var array<string, Definition>
      */
-    private array $workflows = [];
+    private array $definitions = [];
+
+    private ServiceLocator $workflows;
 
     private const DUMP_FORMAT_OPTIONS = [
         'puml',
         'mermaid',
         'dot',
     ];
 
-    public function __construct(array $workflows)
+    public function __construct($workflows)
     {
         parent::__construct();
 
-        $this->workflows = $workflows;
+        if (is_array($workflows)) {
+            $this->definitions = $workflows;
+            trigger_deprecation('symfony/framework-bundle', '6.2', 'Passing an array of definitions in "%s()" is deprecated. Inject a ServiceLocator filled with all workflows instead.', __METHOD__);
+        } elseif ($workflows instanceof ServiceLocator) {
```

Comment:
why not this as first if, as it will be the "new code"

Reply:
Fixed